SU2_CFD/src/numerics/flow/flow_sources.cpp

```diff
@@ -430,6 +432,20 @@ CNumerics::ResidualType<> CSourceIncBodyForce::ComputeResidual(const CConfig* co
   return ResidualType<>(residual, jacobian, nullptr);
 }
 
+CSourceBFM::CSourceBFM(unsigned short val_nDim, unsigned short val_nVar, const CConfig* config) :
+                     CNumerics(val_nDim, val_nVar, config) {
+   BFM_sources.resize(val_nDim + 2);
+
+}
+
+CNumerics::ResidualType<> CSourceBFM::ComputeResidual(const CConfig* config){
+  su2double Force_Ref = config->GetForce_Ref();
+  su2double residual[nDim + 2];
+  for(unsigned short iDim=0; iDim<nDim + 2; ++iDim){
+    residual[iDim] = -Volume * BFM_sources.at(iDim) / Force_Ref;
+  }
+  return ResidualType<>(residual, nullptr, nullptr);
+}
```

Comment:
Your numerics are so simple that I would not use a numerics class, just implement them via your solver for example, that way you can use your feature + any other existing source.